### PR TITLE
PPTP-429: Job title page impl

### PR DESCRIPTION
**Pull request by tombart**

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.forms
+
+trait CommonFormValidators {
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala:19]: :+1:

We will have to update the other form models.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitle.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.forms
+
+import play.api.data.Forms.{mapping, text}
+import play.api.data.{Form, Forms}
+import play.api.libs.json.Json
+
+case class JobTitle(value: Option[String]) {}
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitle.scala:23]: Why are we making `JobTitle` optional? If `JobTitle` is required then this should be mandatory 
This should be sufficient.
```
case class JobTitle(value: String)
```
with Idiomatic Scala and _brevity_ we do not need the curly braces.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitle.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  val maxLength                      = 155
+  val jobTitle                       = "value"
+  lazy val jobTitleEmptyError        = "primaryContactDetails.jobTitle.empty.error"
+  lazy val jobTitleTooLongError      = "primaryContactDetails.jobTitle.tooLong.error"
+  lazy val jobTitleNonAlphaCharError = "primaryContactDetails.jobTitle.nonAlphaChar.error"
+
+  def form(): Form[JobTitle] =
+    Form(
+      mapping(
+        jobTitle -> Forms.optional(text())
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitle.scala:37]: If JobTitle is mandatory then `Forms.optional` will have to be removed.

File: conf/messages.en [hunk 4]
```diff
@@ -97,4 +97,10 @@ primaryContactDetails.fullNamePage.lastName.error.specialCharacters = Enter a la
 
 sessionTimout.title = For your security, we signed you out
 sessionTimout.paragraph.saved = We saved your answers.
-sessionTimout.signin.button = Sign in
\ No newline at end of file
+sessionTimout.signin.button = Sign in
+
+primaryContactDetails.sectionHeader = Primary contact details
```
Comment (PauloPortugal) [inline conf/messages.en:102]: Please delete:
```
primaryContactDetails.sectionHeader = Primary contact details
```

This already exists.

File: test/base/unit/ControllerSpec.scala [hunk 5]
```diff
@@ -76,7 +82,11 @@ trait ControllerSpec
   protected def getTuples(cc: AnyRef): Seq[(String, String)] =
     cc.getClass.getDeclaredFields.foldLeft(Map.empty[String, String]) { (a, f) =>
       f.setAccessible(true)
-      a + (f.getName -> f.get(cc).toString)
+      a + (f.getName -> getValue(f, cc))
     }.toList
 
+  private def getValue(field: Field, cc: AnyRef): String =
```
Comment (PauloPortugal) [inline test/base/unit/ControllerSpec.scala:88]: Ahhh!!!! Awesome! :smile:

File: test/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitleSpec.scala [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import play.api.data.FormError
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.JobTitle.{
+  jobTitle,
+  jobTitleEmptyError,
+  jobTitleNonAlphaCharError,
+  jobTitleTooLongError,
+  maxLength
+}
+
+class JobTitleSpec extends AnyWordSpec with Matchers with CommonTestUtils {
```
Comment (PauloPortugal) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitleSpec.scala:31]: Good point!!! We probably should test the mapping here.

**Review by PauloPortugal — COMMENTED** [review]
Please review the comments

File: conf/messages.en (hunk 4 above)
Comment (tombart) [reply conf/messages.en:102]: done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala (hunk 1 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala:21]: done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitle.scala (hunk 2 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/JobTitle.scala]: done

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**